libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -158,6 +159,90 @@ def map_to_base_relationship(rel: Any) -> Relationship:
     )
 
 
+def _convert_to_graph_document(
+    raw_schema: Dict,
+) -> Tuple[List[Node], Tuple[List[Relationship]]]:
+    # If there are validation errors
+    if not raw_schema["parsed"]:
+        try:
+            argument_json = json.loads(
+                raw_schema["raw"].additional_kwargs["tool_calls"][0]["function"][
+                    "arguments"
+                ]
+            )
+            nodes = []
+            for node in argument_json["nodes"]:
+                if not node.get("id"):  # Id is mandatory
+                    continue
+                nodes.append(
+                    Node(
+                        id=node["id"].title(),
+                        type=node.get("type", "Node").capitalize(),
+                    )
+                )
+            relationships = []
+            for rel in argument_json["relationships"]:
+                # Mandatory props
+                if (
+                    not rel.get("source_node_id")
+                    or not rel.get("target_node_id")
+                    or not rel.get("type")
+                ):
+                    continue
+
+                # Node type copying if needed from node list
+                if not rel.get("source_node_type"):
+                    try:
+                        rel["source_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["source_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["source_node_type"] = "Node"
+                if not rel.get("target_node_type"):
+                    try:
+                        rel["target_node_type"] = [
+                            el["type"]
+                            for el in argument_json["nodes"]
+                            if el["id"] == rel["target_node_id"]
+                        ][0]
+                    except IndexError:
+                        rel["target_node_type"] = "Node"
+
+                source_node = Node(
+                    id=rel["source_node_id"].title(),
```

Comment:
title case or something more id like?
do we do the same changes to the node list too? and shouldn't that be happening later in the processing?

Reply:
It's all consistent, but maybe we could movie it downstream so that it shows up only once in the code